jobs/ServiceInstanceUpdateJob.js

```diff
@@ -86,34 +87,36 @@ class ServiceInstanceUpdateJob extends BaseJob {
         }
         let trackAttempts = true;
         return this
-          .updateDeployment(
-            instanceDetails.deployment_name,
+          .updateServiceInstance(
+            instanceDetails.instance_id,
             diffResults.diff,
             plan.service.force_update,
-            instanceDetails.run_immediately
+            instanceDetails.run_immediately,
+            plan,
+            _.get(resourceDetails, 'context')
           )
           .then(result => {
             operationResponse.update_init = CONST.OPERATION.SUCCEEDED;
-            operationResponse.update_operation_guid = result.guid;
+            operationResponse.update_response = result;
             return operationResponse;
           })
           .catch(err => {
             operationResponse.update_init = CONST.OPERATION.FAILED;
             logger.error('Error occurred while updating service instance job :', err);
-            if (err instanceof errors.DeploymentAttemptRejected) {
+            if (this.deploymentStaggered(err)) {
               //If deployment was staggered due to exhaustion of workers, reschedule update job
               //Retry attempts do not count when deployment is staggered
               //TODO: Need to check if the next run for scheduled update causes problems if the earlier deployment did not go through
               trackAttempts = false;
               err.statusMessage = 'Deployment attempt rejected due to BOSH overload. Update cannot be initiated';
             }
-            if (err instanceof errors.DeploymentAlreadyLocked) {
+            if (this.deploymentLocked(err)) {
```

Comment:
can you also move the lines 119-122 into the method `deploymentLocked` and rename this method into `deploymentLockedForBackup`  and return the boolean accordingly?

Reply:
This is the generic method for 'deploymentLocked'. for backup or not checked in caller

Reply:
Agree but is this deploymentLocked used anywhere else?  If this is common method, then move this to utils so that it can be reused and also move the LN119-120 into another util method `isDeploymentLockedFor(operationName, err)`
```
function isDeploymentLockedFor(operationName, err){
    if(isDeploymentLocked(err)){
          const errMessage = _.get(err, 'error.description') || '';
         const operation = errMessage.split(' ').splice(-1)[0];	              
          if (operation === operationName) {    
                  return true;
          }
       }
   return false;
}
```
Just recommending the above for better readability. 
Also better to name functions with boolean response with prefix `is` or `has`. In this case `is` suits better.

Reply:
moved to utils.